packages/ipfs-http-client/src/add/index.js

```diff
@@ -2,56 +2,48 @@
 
 const ndjson = require('iterable-ndjson')
 const CID = require('cids')
-const configure = require('../lib/configure')
+// const configure = require('../lib/configure')
 const toIterable = require('stream-to-it/source')
 const { toFormData } = require('./form-data')
 const toCamel = require('../lib/object-to-camel')
+const merge = require('merge-options')
 
-module.exports = configure(({ ky }) => {
-  return async function * add (input, options) {
-    options = options || {}
+module.exports = api => {
+  return async function * add (input, options = {}, fetchOptions = {}) {
```

Comment:
Should we stick to the interface api here and not add extra arguments?

I see `timeout`, `signal` and `headers` - `signal` will become a thing with https://github.com/ipfs/interface-js-ipfs-core/issues/58, `timeout` might then be redundant I guess as you can just race them and use the signal to do the timeout.

For headers, I think these are used for authentication mainly when people put a proxy up in front of their nodes, we could allow the user to specify headers added to all requests in the http-client constructor?

Reply:
constructor already supports these 3 and methods 3rd arg merges with the constructor input.

also timeout is already internally connected to any signal passed in to avoid races.

this 3rd argument is totally optional i added it for advanced use cases where you want to customize fetch native options, this is why i didn't re-use the 2nd argument. This should support all these https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch not just headers, timeout and signal.

this should be http-client specific, and if we add signal to the Core API we can easily merge the two.

Reply:
we can even make the constructor consistent with methods adding a `fetchOptions` arg and not mix Core API options with fetch options.

Reply:
I think merging the two would be preferable, otherwise we have embrace-and-extend style implementations of the core interface.

We can think about the fetch args we want to allow through - it's supposed to be an RPC client, exposing too many transport concerns makes the abstraction leak.